tic_tac_toe/game.py

```diff
@@ -87,18 +131,59 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    
+    # Check if board full.
+    if (_board_is_full(game['board'])):
+        raise InvalidMovement('Game is over.')
+        
+    # Check if there is a winner.
+    if game['winner']:
+        raise InvalidMovement('Game is over.')
+
+    # Check that it is this player's turn.
+    if player != game['next_turn']:
+        raise InvalidMovement('"{}" moves next'.format(game['next_turn']))
+
+    # Check if the position is valid.
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+
+    # Check if position is take.
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement('Position already taken.')
+
+    # Execute player move.
+    row, column = position
+    game['board'][row][column] = player
+
+    # Check if tie.
+    if (_board_is_full(game['board'])):
+        raise GameOver('Game is tied!')
+        
+    # Check if winner.
+    if _check_winning_combinations(game['board'], player):
+        game['winner'] = player
+        raise GameOver('"{}" wins!'.format(player))
+
+    # Update the next move.
+    game['next_turn'] = game['player1'] if (player == game['player2']) else game['player2']
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    board = game.get('board')
+    row1 = '{}  |  {}  |  {}'.format(board[0][0], board[0][1], board[0][2])
+    row2 = '{}  |  {}  |  {}'.format(board[1][0], board[1][1], board[1][2])
+    row3 = '{}  |  {}  |  {}'.format(board[2][0], board[2][1], board[2][2])
+    delimeter = '--------------'
+    return '\n' + row1 + '\n' + delimeter + '\n' + row2 + '\n' + delimeter + '\n' + row3 + '\n'
 
 
 def get_next_turn(game):
     """
     Returns the player who plays next, or None if the game is already over.
     """
-    pass
+    if (get_winner(game) is None) and (_board_is_full(game['board']) == False):
+        return game['next_turn']
```

Comment:
Really good code guys! Lots of good ideas our group could learn from. Other than the bug I pointed out above, I would say my only other comment would be that your range(x) functions don't require two arguments since the 0 is implied. I would also say that your variable assignment (such as with board = the game board) is unneeded and takes up unnecessary resources (if I'm not mistaken), but that's really tiny stuff that won't really improve your code very much. Good job!